public/app/features/query/state/updateQueries.ts

```diff
@@ -26,9 +26,15 @@ export async function updateQueries(
     else if (currentDS && nextDS.importQueries) {
       nextQueries = await nextDS.importQueries(queries, currentDS);
     }
-    // Otherwise clear queries
+    // check to see if an existing query for that datasource exists in nextQueries and preserve it if it exists
     else {
-      return [DEFAULT_QUERY];
+      const savedQuery = nextQueries.find((query) => query.datasource?.type === nextDS.type);
```

Comment:
Great catch! I also agree with the fact that if there are multiple queries it will only save the first one - this seems potentially unwanted behavior. I am going to close this particular issue and see if I can tackle it for Azure Monitor exclusively (instead of every datasource). Thank you all so much for your input!